sdk/ai/ai-projects/src/agents/vectorStores.ts

```diff
@@ -55,22 +56,63 @@ export async function listVectorStores(
 }
 
 /** Creates a vector store. */
-export async function createVectorStore(
+export function createVectorStore(
   context: Client,
   options: CreateVectorStoreOptionalParams = {},
-): Promise<VectorStoreOutput> {
+): CreateVectorStoreResponse {
   const createOptions: GeneratedParameters.CreateVectorStoreParameters = {
     ...operationOptionsToRequestParameters(options),
     body: ConvertToWire.convertVectorStoreOptions(options),
   };
-
+  const pollingOptions = convertPollingOptions(options);
   validateCreateVectorStoreParameters(createOptions);
-  const result = await context.path("/vector_stores").post(createOptions);
 
-  if (!expectedStatuses.includes(result.status)) {
-    throw createRestError(result);
+  async function executeCreateVectorStore(): Promise<VectorStoreOutput> {
+    const result = await context.path("/vector_stores").post(createOptions);
+    if (!expectedStatuses.includes(result.status)) {
+      throw createRestError(result);
+    }
+    return ConvertFromWire.convertVectorStoreOutput(result.body);
   }
-  return ConvertFromWire.convertVectorStoreOutput(result.body);
+
+  async function updateCreateVectorStore(
+    currentResult?: VectorStoreOutput,
+  ): Promise<{ result: VectorStoreOutput; completed: boolean }> {
+    let vectorStore: VectorStoreOutput;
+    if (!currentResult) {
+      vectorStore = await executeCreateVectorStore();
+    } else {
+      const getOptions: GetVectorStoreOptionalParams = {
+        ...operationOptionsToRequestParameters(options),
+      };
+      vectorStore = await getVectorStore(context, currentResult.id, getOptions);
+    }
+    return {
+      result: vectorStore,
+      completed: vectorStore.status !== "in_progress",
+    };
+  }
+
+  const poller = new AgentsPoller<VectorStoreOutput>({
+    update: updateCreateVectorStore,
+    pollingOptions: pollingOptions,
+  });
+
+  async function pollOnce(): Promise<VectorStoreOutput> {
+    await poller.poll();
+    const initialResult = poller.getOperationState().result;
+    if (!initialResult) {
+      throw new Error("Error creating vector store");
```

Comment:
How can we throw an error that gives user about why it failed. Also if would poll rethrow the error ?

Reply:
I think that in practice this error should never get thrown because the first call to poll() will set the result but the getOperationState() as defined by the LRO library will either return the state or undefined so I needed to add a check. Is there some behavior instead of throwing an error I should use in the check?